OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1

```diff
@@ -0,0 +1,440 @@
+  return Test-Path $fileName
+}
+
+function New-XmlAttribute($xml, $name, $value) {
+  $attribute = $xml.CreateAttribute($name)
+  $attribute.Value = $value
+  return $attribute
+}
+
+function ConvertTo-HashTable {
+  [CmdletBinding()]
+  [OutputType([HashTable])]
+  param
+  (
+    [Microsoft.Management.Infrastructure.CimInstance[]] $tokens
+  )
+  $HashTable = @{}
+  foreach($token in $tokens) {
+    $HashTable.Add($token.Key, $token.Value)
+  }
+  return $HashTable
+}
+
+function Request-SeqClientNlogDll ($dllPath) {
+  Write-Verbose "Downloading Seq.Client.NLog.dll version 2.3.25 from nuget to $dllPath"
+
+  $ProgressPreference = "SilentlyContinue"
+  $folder = [System.IO.Path]::GetTempPath()
+  Invoke-WebRequest https://dist.nuget.org/win-x86-commandline/latest/nuget.exe -outfile "$folder\nuget.exe"
+  & "$folder\nuget.exe" install Seq.Client.NLog -outputdirectory $folder -version 2.3.24
+  Copy-Item "$folder\Seq.Client.NLog.2.3.24\lib\net40\Seq.Client.NLog.dll" $dllPath
+}
+
+function Save-NlogConfig ($nlogConfig, $filename) {
+  $nlogConfig.Save($filename)
+}
+
+function Test-HashTable($currentValue, $requestedValue) {
+  $currentConfigurationMatchesRequestedConfiguration = $true
+  if ($currentValue.Count -ne $requestedValue.Count) {
+    Write-Verbose "(FOUND MISMATCH) Configuration parameter '$key' with $($currentValue.count) values mismatched the specified $($requestedValue.Count) values"
+    $currentConfigurationMatchesRequestedConfiguration = $false
+  } else {
+    foreach($value in $currentValue.Keys) {
+      $curr = $currentValue[$value]
+      $req = $requestedValue[$value]
+      if ($curr -ne $req)
+      {
+        Write-Verbose "(FOUND MISMATCH) Configuration parameter `"$key['$value']`" with value '$curr' mismatched the specified value '$req'"
+        $currentConfigurationMatchesRequestedConfiguration = $false
+      }
+    }
+  }
+  return $currentConfigurationMatchesRequestedConfiguration
+}
+
+function Test-PSCredential($currentValue, $requestedValue) {
+  if ($null -ne $currentValue) {
+    $currentUsername = $currentValue.GetNetworkCredential().UserName
+    $currentPassword = $currentValue.GetNetworkCredential().Password
+  } else {
+    $currentUserName = ""
+    $currentPassword = ""
+  }
+
+  if ($null -ne $requestedValue) {
+    $requestedUsername = $requestedValue.GetNetworkCredential().UserName
+    $requestedPassword = $requestedValue.GetNetworkCredential().Password
+  } else {
+    $requestedUsername = ""
+    $requestedPassword = ""
+  }
+
+  if ($currentPassword -ne $requestedPassword -or $currentUsername -ne $requestedUserName) {
+    Write-Verbose "(FOUND MISMATCH) Configuration parameter '$key' with value '********' mismatched the specified value '********'"
+    return $false
+  }
+  else {
+    Write-Verbose "Configuration parameter '$key' matches the requested value '********'"
+  }
+  return $true
+}
+
+function Get-OctopusDSCParameter($parameters) {
```

Comment:
I feel like we should have a shared code module for bits like this. I'll open a separate isue for that.

Reply:
We should, yeah. Lots of common code. Even the implementation of `Test-TargetResource` is the same across most resources